dial_test.go

```diff
@@ -480,3 +480,30 @@ func TestDialBackoffClears(t *testing.T) {
 		t.Log("correctly cleared backoff")
 	}
 }
+
+func TestDialPeerFailed(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	swarms := makeSwarms(ctx, t, 2)
+	defer closeSwarms(swarms)
+	testedSwarm, targetSwarm := swarms[0], swarms[1]
+
+	for i := 0; i < 10; i++ {
+		_, silentPeerAddress, silentPeerListener := newSilentPeer(t)
+		go acceptAndHang(silentPeerListener)
+		defer silentPeerListener.Close()
+
+		testedSwarm.Peerstore().AddAddr(
+			targetSwarm.LocalPeer(),
+			silentPeerAddress,
+			pstore.PermanentAddrTTL)
+	}
+
+	_, err := testedSwarm.DialPeer(ctx, targetSwarm.LocalPeer())
+	if err == nil {
+		t.Fatal(err)
+	}
+
+	t.Logf("correctly get a combined error: %s", err)
```

Comment:
This should test that we're getting the _right_ errors.

Reply:
I thought about it. But was not sure about check. The only way I see is to check err.Error() content itself. Is it ok?

Reply:
Sounds good. Unfortunately, that's usually how error checking works in go. Ideally, there'd be two different errors to test for.

Reply:
I have added the additional checks for Error() content.